File: app/src/main/java/br/edu/farol/gadoplus/adapter/LoteAdapter.java
package br.edu.farol.gadoplus.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import br.edu.farol.gadoplus.R;
import br.edu.farol.gadoplus.model.Lote;

public class LoteAdapter extends RecyclerView.Adapter<LoteAdapter.LoteViewHolder> {

    private List<Lote> lotes;
    private OnItemClickListener listener;


    @Override
    public LoteViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recyclerview_item, parent, false);

        return new LoteViewHolder(itemView);
    }


    @Override
    public void onBindViewHolder(LoteViewHolder viewHolder, int i) {

        viewHolder.nome.setText(lotes.get(i).getNome());
        viewHolder.descricao.setText(lotes.get(i).getDescricao());

    }

    @Override
    public int getItemCount() {
        if (lotes != null)
            return lotes.size();
        else return 0;
    }

    public void setLote(List<Lote> lotes) {
        this.lotes = lotes;
        notifyDataSetChanged();
    }

    class LoteViewHolder extends RecyclerView.ViewHolder {

        private TextView nome, descricao, sPropriedade;

        public LoteViewHolder(View itemView) {
            super(itemView);
            nome = itemView.findViewById(R.id.text_view_title);
            descricao = itemView.findViewById(R.id.text_view_description);
            sPropriedade = itemView.findViewById(R.id.text_view_priority);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    int position = getAdapterPosition();
                    if (listener != null && position != RecyclerView.NO_POSITION)
                        listener.onItemClick(lotes.get(position));
                }
            });
        }
    }

    public interface OnItemClickListener {
        void onItemClick(Lote lote);
    }

    public void setOnClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }
    
}

File: app/src/main/java/br/edu/farol/gadoplus/adapter/LoteAdapter.kt
package br.edu.farol.gadoplus.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.edu.farol.gadoplus.R
import br.edu.farol.gadoplus.adapter.LoteAdapter.LoteViewHolder
import br.edu.farol.gadoplus.model.Lote

class LoteAdapter : RecyclerView.Adapter<LoteViewHolder>() {
    private var lotes: List<Lote>? = null
    private var listener: OnItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LoteViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.recyclerview_item, parent, false)
        return LoteViewHolder(itemView)
    }

    override fun onBindViewHolder(viewHolder: LoteViewHolder, i: Int) {
        viewHolder.nome.text = lotes!![i].nome
        viewHolder.descricao.text = lotes!![i].descricao
    }

    override fun getItemCount(): Int {
        return if (lotes != null) lotes!!.size else 0
    }

    fun setLote(lotes: List<Lote>?) {
        this.lotes = lotes
        notifyDataSetChanged()
    }

    inner class LoteViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nome: TextView
        val descricao: TextView
        private val sPropriedade: TextView

        init {
            nome = itemView.findViewById(R.id.text_view_title)
            descricao = itemView.findViewById(R.id.text_view_description)
            sPropriedade = itemView.findViewById(R.id.text_view_priority)
            itemView.setOnClickListener {
                val position = adapterPosition
                if (listener != null && position != RecyclerView.NO_POSITION) listener!!.onItemClick(lotes!![position])
            }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(lote: Lote?)
    }

    fun setOnClickListener(listener: OnItemClickListener?) {
        this.listener = listener
    }
}